app/main.py

```diff
@@ -1 +1,18 @@
-# write your code here
+from app.errors import VaccineError, NotWearingMaskError
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_needed = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_needed += 1
+
+    if masks_needed != 0:
+        return f"Friends should buy {masks_needed} masks"
+
+    return f"Friends can go to {cafe.name}"
```

Comment:
The return statement should be `return f"Welcome to {cafe.name}"` instead of `return f"Friends can go to {cafe.name}"` to meet the task requirements.